spec/system/dashboards/user_sorts_dashboard_articles_spec.rb

```diff
@@ -5,6 +5,22 @@
   let(:article1) { create(:article, user_id: user.id, published_at: 10.minutes.ago, created_at: 1.day.ago, positive_reactions_count: 5, page_views_count: 0, comments_count: 100) }
   let(:article2) { create(:article, user_id: user.id, published_at: 1.minute.ago, created_at: 2.days.ago, positive_reactions_count: 1, page_views_count: 10, comments_count: 0) }
   let(:article3) { create(:article, user_id: user.id, published_at: 5.minutes.ago, created_at: 3.days.ago, positive_reactions_count: 0, page_views_count: 1000, comments_count: 50) }
+  let(:articles) { [article1, article2, article3] }
+
+  let(:article_with_comments_count_of) do
+    ->(target_count) { articles.detect { |article| article.comments_count == target_count } }
```

Comment:
This looks great! 

Small little optimization suggestion. If we were working with a bigger array here looping through the entire array every time would be very slow. A faster approach would be to use a hash with the comment counts as the keys. 

```ruby
articles = { article1.comments_count => article1, ... }
->(target_count) { articles[target_count] }
```

We don't need to do that here bc this array is small but something thats good to keep in mind if you find yourself with a very large array 😊

Reply:
Agreed, and thanks for the kind words!